File: software/examples/perl/example_loopback.pl
#!/usr/bin/perl

# For this example connect the RX+/- pins to TX+/- pins on the same Bricklet
# and configure the DIP switch on the Bricklet to full-duplex mode

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletRS485;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your RS485 Bricklet

# Callback subroutine for read callback
sub cb_read
{
    my ($message) = @_;

    print "Message: \"" . join('', @{$message}) . "\"\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $rs485 = Tinkerforge::BrickletRS485->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Enable full-duplex mode
$rs485->set_rs485_configuration(115200, $rs485->PARITY_NONE, $rs485->STOPBITS_1,
                                $rs485->WORDLENGTH_8, $rs485->DUPLEX_FULL);

# Register read callback to subroutine cb_read
$rs485->register_callback($rs485->CALLBACK_READ, 'cb_read');

# Enable read callback
$rs485->enable_read_callback();

# Write "test" string
my @message = split('', 'test');
$rs485->write(\@message);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/php/ExampleLoopback.php
<?php

// For this example connect the RX+/- pins to TX+/- pins on the same Bricklet
// and configure the DIP switch on the Bricklet to full-duplex mode

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletRS485.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletRS485;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your RS485 Bricklet

// Callback function for read callback
function cb_read($message)
{
    echo "Message: \"" . implode('', $message) . "\"\n";
}

$ipcon = new IPConnection(); // Create IP connection
$rs485 = new BrickletRS485(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Enable full-duplex mode
$rs485->setRS485Configuration(115200, BrickletRS485::PARITY_NONE,
                              BrickletRS485::STOPBITS_1, BrickletRS485::WORDLENGTH_8,
                              BrickletRS485::DUPLEX_FULL);

// Register read callback to function cb_read
$rs485->registerCallback(BrickletRS485::CALLBACK_READ, 'cb_read');

// Enable read callback
$rs485->enableReadCallback();

# Write "test" string
$rs485->write(str_split('test'));

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
